config/config.hpp

```diff
@@ -30,11 +30,18 @@
 #endif
 
 // Windows specific options
-#ifdef _WIN32
-#ifndef _USE_MATH_DEFINES
+#if defined(_WIN32) && !defined(_USE_MATH_DEFINES)
 // Macro needed to get defines like M_PI from <cmath>. (Visual Studio C++ only?)
 #define _USE_MATH_DEFINES
```

Comment:
Not sure if it's necessary, but axom also defines the following on Windows:
```cpp
#define NOGDI
#define NOMINMAX
```
https://github.com/LLNL/axom/blob/f30c0a06ce91e5749202f6db91fec12c0b3460e2/src/axom/config.hpp.in#L89-L93

Reply:
I don't know what these do. Do you know why they are added in Axom?

Reply:
I think some Windows header files add a few defines/macros that pollute the global namespace.

* `NOMINMAX` prevents windows from a `#define min(a,b)` and a `#define max(a,b)` macros
* I had to look up the `NOGDI` one (the note in axom says it was added to disambiguate geometry classes). Among other things, I think it adds a macro for `#define ERROR 0`. There are probably others, and I'm not sure if it would help w/ https://github.com/mfem/mfem/issues/3451, but it might be worth checking.

Reply:
Since we do not include `windows.h` from `mfem.hpp` (or any other header), I don't think we need to include these macros.

If a user includes `windows.h` and `mfem.hpp` they may need to define these before including `windows.h`. Also, if they include `windows.h` before `mfem.hpp` then our definitions (if we included these macros) won't do anything anyway.

We actually use `NOMINMAX` locally in a `.cpp` file, here: https://github.com/mfem/mfem/blob/ea5c4664864cb84c0a453f527d6befec05a8b8a9/general/tic_toc.cpp#L26-L28
where we include `windows.h` for `QueryPerformanceFrequency` and `QueryPerformanceCounter`.

Reply:
+1 for C++11 timers making life much easier for multi-platform.

Reply:
`MPI_Wtime` is also not bad for cross-platform. The options we have are all from pre- C++11 times. We can probably switch to C++11 but I'm not sure if it uses the high-res option on all platforms.